edx_exams/settings/test.py

```diff
@@ -23,3 +23,47 @@
 )
 
 LMS_ROOT_URL = 'http://test.lms:18000'
+
+TOKEN_SIGNING = {
```

Comment:
we'll need to have this setting setup to work in other environments too. Typically what we'd do is put this in base.py and than override it in the local/test/production files. We can take a look at this after some of the other comments above tho since it's just config.

As an example https://github.com/edx/edx-exams/blob/2ca064b98a5c77f40b9583d1ab7147ad250bb499/edx_exams/settings/base.py#L218

Reply:
@Zacharis278 I see, good point. So I had grabbed the TOKEN_SIGNING from the token-utils repo: https://github.com/edx/token-utils/blob/main/test_settings.py#L64

But these are specifically only used for the test settings. For base.py, should I set these to none and then have them overridden with something else for local/production? Specifically wondering about JWT_PUBLIC_SIGNING_JWK_SET and JWT_PRIVATE_SIGNING_JWK as the other stuff looks like it can be left as is.

Reply:
ok think I've got my head around this, base.py should include the token settings we always need so something like this

```
TOKEN_SIGNING = {
    'JWT_ISSUER': 'http://127.0.0.1:8740/',
    'JWT_SIGNING_ALGORITHM': 'RS512',
    'JWT_SUPPORTED_VERSION': '1.2.0',
    'JWT_PRIVATE_SIGNING_JWK': 'replace-me',
}
```

To deal with our dev environments we'll just update JWT_PRIVATE_SIGNING_JWK to have a value in local.py. (see that local.py imports from base.py)

Notice I didn't include the JWT_PUBLIC_SIGNING_JWK_SET. edx-exams doesn't need to validate this JWT so it doesn't need that setting however we should include it in test.py. So in test we'll need to do an update on both JWT_PRIVATE_SIGNING_JWK and JWT_PUBLIC_SIGNING_JWK_SET.

Finally for prod/stage we need to have TOKEN_SIGNING included in https://github.com/edx/edx-exams/blob/b02c4f373471e0ad175a56ddb4c96c121008513e/edx_exams/settings/production.py#L17 so we can add our settings to edx-internal.